clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -256,7 +268,7 @@ public Cluster cluster() {
             }
         }
 
-        return new Cluster(this.brokers, partitions, unauthorizedTopics);
+        return new Cluster(this.brokers, partitions, topicsByError(Errors.TOPIC_AUTHORIZATION_FAILED));
```

Comment:
If we use the newly introduced method, then we should remove the code that populates `unauthorizedTopics`.